File: cms_web/src/Services/Entities/DeptResidentEntity.ts
import DepartmentEntity from "./DepartmentEntity";
import HospResidentEntity from "./HospResidentEntity";

interface DeptResidentEntity {
  dept_res_pk?: number;
  dept_pk?: number;
  res_pk?: number;
  notes?: string;
  is_active?: "y" | "n";
  encoded_at?: Date;
  encoder_pk?: string;

  department?: DepartmentEntity;
  hosp_res?: HospResidentEntity;
}

export interface DeptResidentTableModel {
  limit: number;
  count: number;
  begin: number;
  table: Array<DeptResidentEntity>;
}

export default DeptResidentEntity;

File: cms_server/cms_server/Entities/DeptResidentEntity.cs
using System;

namespace cms_server.Entities
{
    public class DeptResidentEntity
    {
        public int dept_res_pk { get; set; }
        public int dept_pk { get; set; }
        public int res_pk { get; set; }
        public string notes { get; set; }
        public string is_active { get; set; }
        public DateTime? encoded_at { get; set; }
        public string encoder_pk { get; set; }

        //ext
        public DepartmentEntity department { get; set; }
        public HospResidentEntity hosp_res { get; set; }
    }
}
